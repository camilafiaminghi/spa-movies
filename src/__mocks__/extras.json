[
	{
		"type": "rate",
		"author": "",
		"title": "5/5",
		"subtitle": "★★★★★",
		"description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit.",
		"icon": "",
		"date": ""
	},

	{
		"type": "rate",
		"author": "",
		"title": "14+",
		"subtitle": "Lorem ipsum",
		"description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit.",
		"icon": "images/icon/checked.png",
		"date": ""
	},

	{
		"type": "rate",
		"author": "",
		"title": "64%",
		"subtitle": "Lorem ipsum",
		"description": [["116","Reviews"], ["74", "Fresh"], ["42", "Rotten"], ["5.7", "Average"]],
		"icon": "images/icon/tomato.png",
		"date": ""
	},

	{
		"type": "comment",
		"author": "Lorem ipsum",
		"title": "Lorem",
		"subtitle": "",
		"description": "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
		"icon": "images/icon/splash.png",
		"date": "Jul 29 2016"
	},

	{
		"type": "comment",
		"author": "Kely Smith",
		"title": "Lorem",
		"subtitle": "",
		"description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt",
		"icon": "images/icon/tomato.png",
		"date": "Jul 29 2016"
	},

	{
		"type": "comment",
		"author": "Lorem ipsum",
		"title": "Lorem",
		"subtitle": "",
		"description": "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
		"icon": "images/icon/splash.png",
		"date": "Jul 29 2016"
	},

	{
		"type": "comment",
		"author": "Kely Smith",
		"title": "Lorem",
		"subtitle": "",
		"description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt",
		"icon": "images/icon/tomato.png",
		"date": "Jul 29 2016"
	}
]
