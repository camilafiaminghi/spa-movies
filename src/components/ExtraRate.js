import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

const ExtraRate = ({title, subtitle, icon, description}) => {

	return (
		<Fragment>
			<h1>
				{(icon) && <img className="icon" src={require(`./../${icon}`)} alt={title} />} {title}
			</h1>
			<h2 className="h4">{subtitle}</h2>
			{(typeof description === 'string') && <p>{description}</p>}
			{(typeof description === 'object') && (
				<dl>
					{description.map((item, index) => (
						<Fragment key={`description-item-${index}`}>
							<dt>{item[0]}</dt>
							<dd>{item[1]}</dd>
						</Fragment>
					))}
				</dl>
			)}
		</Fragment>
	)
}

ExtraRate.propTypes = {
	title: PropTypes.string,
	subtitle: PropTypes.string,
	icon: PropTypes.string,
	description: PropTypes.oneOfType([
	  PropTypes.string,
	  PropTypes.array
	])
}

export default ExtraRate
