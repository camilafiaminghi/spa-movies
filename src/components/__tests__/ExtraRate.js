import React from 'react'
import {
	render,
	cleanup
} from '@testing-library/react'
import ExtraRate from './../ExtraRate'

import extras from './../../__mocks__/extras.json'
const item = extras.filter((item) => (item.type === 'rate'))[1]

describe('<ExtraRate />', () => {

	afterEach(cleanup)

	it('renders the component', () => {
		const container = render(<ExtraRate {...item} />)
		expect(container.firstChild).toMatchSnapshot()
	})

	it('has an img tag, description, title, subtitle', () => {
		const { getByAltText, getByText } = render(<ExtraRate {...item} />)

		expect(getByAltText(item.title)).toBeDefined()
		expect(getByText(item.title)).toBeDefined()
		expect(getByText(item.subtitle)).toBeDefined()
		expect(getByText(item.description)).toBeDefined()
	})
})
