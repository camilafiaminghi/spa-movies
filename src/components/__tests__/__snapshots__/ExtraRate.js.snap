// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtraRate /> renders the component 1`] = `undefined`;
